### ISSUE TITLE
fixes for v1.18.0

### ISSUE THREAD

**Issue by romange** [dragonflydb/dragonfly#2953]
1. change `table_growth_margin` to 0.4 
2. expose oom_errors via /metrics
more fixes  for `info_replication_valkey_compatible flag:

3. 
```
Master:
[127.0.0.1:6379](http://127.0.0.1:6379/)> role
1) "master"
2) 1) 1) "172.23.72.37"
      2) "6379"
      3) "stable_sync"
[127.0.0.1:6379](http://127.0.0.1:6379/)>
[127.0.0.1:6379](http://127.0.0.1:6379/)> info replication
# Replication
role:master
connected_slaves:1
slave0:ip=172.23.72.37,port=6379,state=stable_sync,lag=0
master_replid:3b210550e4779dcebc7ad9f34c0288afe1b5234f
```
`stable_sync` must be changed to `online`

**Comment by kostasrim** [comment]
Why are we replacing `stable_sync` with `online` ?

**Comment by adiholden** [comment]
@romange we have oom_errors_total in metrix

**Comment by adiholden** [comment]
@romange for number 3 , do you mean drop the flag info_replication_valkey_compatible ?

**Comment by romange** [comment]
not, drop  - improve it. seems we do not cover all the cases for this flag and `stable_sync` is still leaked on master side.

**Comment by adiholden** [comment]
fixed by #2936
